### PR TITLE
Load history when starting a direct debug session

### PR DESCRIPTION
**Pull request by sunblaze**
When starting a debug session directly with RUBY_DEBUG_IRB_CONSOLE=1 and `require 'debug'; debugger`, IRB's history wasn't loaded. This commit ensures history is loaded in this case by calling `load_history` when configuring IRB for the debugger.

Fixes ruby/irb#975

### CODE REVIEW

File: test/irb/test_history.rb [hunk 1]
```diff
@@ -488,6 +488,28 @@ def foo
...
+        debugger
+        binding.irb # needed to satisfy run_ruby_file
+      RUBY
+
+      output = run_ruby_file do
+        type "history"
+        type "exit!"
+      end
+
+      assert_include(output, "old_history_3")
```
Comment (st0012) [inline test/irb/test_history.rb:510]: The current test already captures what we want to cover, but I think we can make expectations even clearer to help future readers understand the context:

```suggestion
      output = run_ruby_file do
        type "history"
        type "puts 'foo'"
        type "history"
        type "exit!"
      end

      assert_include(output, "irb:rdbg(main):002") # assert that we're in an irb:rdbg session
      assert_include(output, "5: history")
      assert_include(output, "4: puts 'foo'")
      assert_include(output, "3: history")
      assert_include(output, "2: old_history_3")
      assert_include(output, "1: old_history_2")
      assert_include(output, "0: old_history_1")
```

**Review by st0012 — APPROVED** [review]
Can confirm the patch fixes the issue, thank you!

File: lib/irb/debug.rb [hunk 2]
```diff
@@ -81,6 +81,7 @@ def configure_irb_for_debugger(irb)
         IRB.instance_variable_set(:@debugger_irb, irb)
         irb.context.with_debugger = true
         irb.context.irb_name += ":rdbg"
+        irb.context.io.load_history
```
Comment (st0012) [inline lib/irb/debug.rb:84]: It looks like we need some safeguarding here as `context.io` doesn't always include `HistorySavingAbility`, especially in some non-tty environments.

```suggestion
        irb.context.io.load_history if irb.context.io.class < HistorySavingAbility
```

File: test/irb/test_history.rb (hunk 1 above)
Comment (sunblaze) [reply test/irb/test_history.rb]: Nice additions!

**Pull request merged**